### ISSUE TITLE
[feature request] Should support ceil and floor functions

### ISSUE THREAD

**Issue by pauldix** [influxdata/influxdb#3691]
Some users want to round results from math and aggregates. These functions should act like transformation functions (like derivative). For example say you have a series where you're reporting uptime in seconds for devices and you want to show how many days it has been up.

```sql
-- ceil rounds up, floor rounds down
-- the first argument is the thing you're taking the ceiling of
-- this would give you as many data points as you have uptime reports, many of which
-- would be the same because of the large divisor
select ceil(value / 86400) from uptime where time > now() - 30d

-- if you're reporting uptime frequently, you may want to limit the number of points here
-- this would give you only 30 data points back
select ceil(last(value) / 86400) from uptime where time > now() - 30d group by time(1d) 

-- ceil and floor should support an optional second argument that specifies the precision to round to
select ceil(value / 86400, .2f) from uptime where time > now() - 30d
```

Just my initial thoughts on this. Feedback welcome.

**Comment by laf** [comment]
Thanks @pauldix 

Exactly what we're looking for :)

**Comment by musl** [comment]
@pauldix Great idea. This will make some history graphs easier to read - especially for those targeted at an audience that's not expected to be well versed in statistics. I think it'll also help with clarity and style in other cases.

**Comment by toddboom** [comment]
Also adding a request for `round()` in response to #5435.

**Comment by beckettsean** [comment]
As mentioned in my [post to the mailing list](https://groups.google.com/d/msg/influxdb/CkKzZoXDgiU/5E-vDNVGCwAJ) we are experimenting with simplifying our open GitHub Issues. This feature request has been rolled into an [aggregate issue](https://github.com/influxdata/influxdb/issues/5930) for all function requests, so that we can close this issue until we are ready to work on it. 

You may continue to make comments here. Closing the issue does not mean we are rejecting this idea.

**Comment by MikiDi** [comment]
Hi,
I would like to have a go at implementing the `CEIL()` and `FLOOR()` functions.
However, I'm new to the project ... could anyone give some pointers on what to look at first, which special cases to consider, etc? :-)

Michaël

**Comment by timhallinflux** [comment]
closed via #9632
